src/shell.zig

```diff
@@ -596,12 +596,33 @@ pub fn spawn_options(
     return child;
 }
 
+pub fn zig_exe_alloc(shell: Shell, allocator: std.mem.Allocator) ![]const u8 {
+    _ = shell;
+
+    // ZIG_EXE is set in `build.zig`, so it is available when using the run step for `zig build
+    // scripts`.
+    // ZIG_EXE is already an absolute path, but zig/zig is not.
+    const zig_exe_default = comptime "zig/zig" ++ builtin.target.exeFileExt();
```

Comment:
So, let's assume that it's just always set!

Reply:
As a bonus to this -- `b.zig_exe` (in `build.zig`) is already an absolute path!